client/main.lua

```diff
@@ -638,6 +638,23 @@ end)
 
 RegisterNetEvent('hospital:client:RespawnAtHospital', function()
     local hospitalIndex = 1 -- Default hospital to respawn at
+    if Config.RespawnAtNearestHospital and #Config.Locations["hospital"] > 0 then
+        local closestH = nil
+        local minDist = nil
+        for i=1, #Config.Locations["hospital"] do
+            if closestH then
+                local newDist = Vdist(Config.Locations["hospital"][i]["location"].x, Config.Locations["hospital"][i]["location"].y, Config.Locations["hospital"][i]["location"].z, GetEntityCoords(PlayerPedId()))
+                if newDist < minDist then
+                    closestH = i
+                    minDist = newDist
+                end
+            else
+                closestH = i
+                minDist = Vdist(Config.Locations["hospital"][i]["location"].x, Config.Locations["hospital"][i]["location"].y, Config.Locations["hospital"][i]["location"].z, GetEntityCoords(PlayerPedId()))
+            end
+        end
+        hospitalIndex = closestH
+    end
```

Comment:
```suggestion
    if Config.RespawnAtNearestHospital and #Config.Locations["hospital"] > 0 then
        local closestH = nil
        local minDist = nil
        local playerPed = PlayerPedId()
        
        if playerPed > 0 and DoesEntityExist(playerPed) then
            local playerCoords = GetEntityCoords(PlayerPedId())
    
            for i=1, #Config.Locations["hospital"] do
                local dist = #(Config.Locations["hospital"][i]["location"] - playerCoords)
                
                if closestH == nil or dist < minDist then
                    closestH = i
                    minDist = dist
                end
            end
        end
        
        if closestH ~= nil then
            hospitalIndex = closestH
        end
    end
```

We should make sure the player actually exists. Even though it should, you can never really be too certain.
The loop can also be simplified, instead of writing the same code twice.

We shouldnt update the `hospitalIdx` if `closestH` is `nil` either

We only need to fetch the player position once, too.